### PR TITLE
Added ship status,time parked and island in /user/ships/

### LINKED ISSUE #56: Add current status details in /user/ships/ endpoint
Add the following details:

- Status- Parked/Not parked
- Island- where it is parked
- Time- when parked

### PR DESCRIPTION
**Pull request by apoorvaeternity**
Added the following details:

- Status-  #Parked/Not parked
- Island- where it is parked
- Time- when parked

Fixes #56

### CODE REVIEW

File: core/serializers.py [hunk 1]
```diff
@@ -258,10 +258,28 @@ class Meta:
 
 class ShipsListSerializer(serializers.ModelSerializer):
     name = serializers.ReadOnlyField(source='ship_store.name')
+    status = serializers.SerializerMethodField('get_ship_status')
+    island_id = serializers.SerializerMethodField()
+    park_time = serializers.SerializerMethodField()
+
+    def get_ship_status(self,obj):
+        if DockChart.objects.filter(ship_id=obj, end_time=None).exists():
+            return "Parked"
```
Comment (shobhitagarwal1612) [inline core/serializers.py]: Status should be "idle" or "busy"

File: core/tests.py [hunk 2]
```diff
@@ -402,7 +402,38 @@ def test_port_details(self):
...
+        Profile.objects.create_player(username='some_username')
+
+        user2 = User.objects.create_user(username='some_username2', password='some_password',
+                                         email='some_email2@gmail.com')
+        Profile.objects.create_player(username='some_username2')
+        ship_id = Ship.objects.get(user=user2).id
+        port_owner_id = Port.objects.filter(user=user).first().user_id
+        port_id = Port.objects.filter(user=user).first().id
+        # Parking user2's raft on user's port
+        data = {'ship_id': ship_id, 'port_owner_id': port_owner_id, 'port_id': port_id}
```
Comment (shobhitagarwal1612) [inline core/tests.py]: This is old code. Have you rebased with the dev branch???

**Comment by apoorvaeternity** [comment]
We should also add:
port_id-The port where the ship is parked.
username-The username of the user on whose port the ship is parked.

File: core/serializers.py [hunk 3]
```diff
@@ -258,10 +258,39 @@ class Meta:
 
 class ShipsListSerializer(serializers.ModelSerializer):
     name = serializers.ReadOnlyField(source='ship_store.name')
+    status = serializers.SerializerMethodField('get_ship_status')
+    island_id = serializers.SerializerMethodField()
+    park_time = serializers.SerializerMethodField()
+    port_id = serializers.SerializerMethodField()
+    username = serializers.SerializerMethodField()
+    def get_ship_status(self, obj):
```
Comment (utkbansal) [inline core/serializers.py]: PEP8

**Comment by apoorvaeternity** [comment]
ADDED
port_id-The port where the ship is parked.
username-The username of the user on whose port the ship is parked.

**Pull request merged**